File: projetos-fenix-para-estudo/t2ti-erp-fenix-c-sharp-com-nhibernate/fontes/backend/c-sharp/T2TiERPFenix-NHibernate/T2TiERPFenix/Models/Transiente/Filtro.cs
/**
 * Classe transiente para controlar o filtro que vem do cliente.
 * tomar como base o seguinte esquema para criar as condições de filtro (filter conditions)
 * https://github.com/nestjsx/crud/wiki/Requests
 */
namespace T2TiERPFenix.Models
{
    public class Filtro
    {
        public Filtro(string filter)
        {
            string[] condicoes = filter.Split("||");

            // $cont (LIKE %val%, contains)
            if (condicoes[1] == "$cont")
            {
                Campo = condicoes[0];
                Valor = condicoes[2];
            }

            // $between (BETWEEN 'DATA1' and 'DATA2')
            if (condicoes[1] == "$between")
            {
                Campo = condicoes[0];
                string[] datas = condicoes[2].Split(",");
                DataInicial = datas[0];
                DataFinal = datas[1];
            }

        }

        public string Campo { get; set; }
        public string Valor { get; set; }
        public string DataInicial { get; set; }
        public string DataFinal { get; set; }

        public string Where {
            set { }

            get
            {
                string where;
                // verifica se o valor é diferente de vazio e se for filtra por campo/valor
                if (!Valor.Equals(""))
                {
                   where = Campo + " like '%" + Valor + "%'";
                }
                // se o valor for vazio, filtra pelo período
                else
                {
                    where = Campo + " between " + DataInicial + " and " + DataFinal;
                }

                return where;
            }

        }
    }
}

File: projetos-fenix-para-estudo/t2ti-erp-fenix-servidores-provider-inicio/fontes/backend/java/fenix/src/main/java/com/t2ti/fenix/model/transiente/Filtro.java
package com.t2ti.fenix.model.transiente;

import java.io.Serializable;

/**
 * Classe transiente para controlar o filtro que vem do cliente.
 * tomar como base o seguinte esquema para criar as condições de filtro (filter conditions)
 * https://github.com/nestjsx/crud/wiki/Requests
 */
public class Filtro implements Serializable {
	private static final long serialVersionUID = 1L;

	private String campo;
	private String valor;
	private String dataInicial;
	private String dataFinal;
	private String where;

	public Filtro(String filter) {
		String filtro = filter.replace("||", "-");
        String[] condicoes = filtro.split("-");

        // $cont (LIKE %val%, contains)
        if (condicoes[1].equals("$cont")) {
            this.campo = condicoes[0];
            this.valor = condicoes[2];
        }		
	}

	public String getCampo() {
		return campo;
	}

	public void setCampo(String campo) {
		this.campo = campo;
	}

	public String getValor() {
		return valor;
	}

	public void setValor(String valor) {
		this.valor = valor;
	}

	public String getDataInicial() {
		return dataInicial;
	}

	public void setDataInicial(String dataInicial) {
		this.dataInicial = dataInicial;
	}

	public String getDataFinal() {
		return dataFinal;
	}

	public void setDataFinal(String dataFinal) {
		this.dataFinal = dataFinal;
	}

	public String getWhere() {
	    // verifica se o valor é diferente de vazio e se for filtra por campo/valor
	    if(!valor.isEmpty()) {
	        where = campo + " like '%" + valor + "%'";
	    }
		// se o valor for vazio, filtra pelo período
		else {
	        where = campo + " between " + dataInicial + " and " + dataFinal;
	    }

	    return where;
	}

	public void setWhere(String where) {
		this.where = where;
	}

}